DoSomething/rogue
# Save tags to db

#### What's this PR do?
- Saves data to database when an admin adds or removes a tag from a post.
- `/tags` endpoint now returns post object with related tags, signup, and user instead of just a status code.

#### How should this be reviewed?
- Make sure all tests still pass! 
- Tag a post.
- Check database to see that it is logged in the `tagging_tagged` table and make sure the tag turns blue.
- Check the events table. The most recent event should have been this tag. 
- Remove the tag from the same post. 
- Check database to see that it has been removed from the `tagging_tagged` table and that the tag is no longer blue.

#### Any background context you want to provide?
I noticed that when you untag a post, an event is not created. I assume that we want this! I also noticed that [laravel-tagging](https://github.com/rtconner/laravel-tagging) has it's own tagging events built in. Do we want to use this? I think this is outside the scope of this PR so @ngjo maybe we can discuss with the team/add a new card for this? 

Also, @DFurnes when I tried to pass a status code to `response()->json($taggedPost);` it broke when untagging something :-/ I think we will need this for events to be able to see if the post has been tagged or untagged and to differentiate between the two events so just wanted to note that here! 

#### Relevant tickets
Fixes #261 

#### Checklist
